src/riscv_machine.cpp

```diff
@@ -235,6 +237,95 @@ static void host_write(void *opaque, uint32_t offset, uint32_t val, int size_log
   }
 }
 
+mc_fifo_t *host_to_mc_req_fifo;
+mc_fifo_t *mc_to_host_req_fifo;
+mc_fifo_t *mc_to_host_resp_fifo;
+
+void manycore_init(RISCVMachine* m) {
+  if (m->manycore) {
+    // Initialize Host --> MC Request FIFO
+    host_to_mc_req_fifo = new mc_fifo_t;
+    mc_fifo_init(host_to_mc_req_fifo);
+
+    // Initialize Host --> MC Request FIFO
+    mc_to_host_req_fifo = new mc_fifo_t;
+    mc_fifo_init(mc_to_host_req_fifo);
+
+    // Initialize Host --> MC Request FIFO
+    mc_to_host_resp_fifo = new mc_fifo_t;
+    mc_fifo_init(mc_to_host_resp_fifo);
+  }
+}
+
+static uint32_t manycore_read(void *opaque, uint32_t offset, int size_log2) { 
+  RISCVMachine *m = (RISCVMachine *)opaque;
+  uint32_t c = 0xFFFFFFFF;
+  if (m->manycore) {
+    switch (offset & 0x0f000) {
```

Comment:
is this assuming some particular addressing scheme?

Reply:
Yes, the addressing scheme is defined here - https://github.com/sripathi-muralitharan/dromajo/blob/f198a4272c633795fbebd01dac4991cc3c32e252/include/riscv_machine.h#L139-L148. I can add a comment pointing to this file for future reference, when adding comments for the other functions in this file.